controllers/users.js

```diff
@@ -738,11 +738,34 @@ const filterUsers = async (req, res) => {
       return res.boom.badRequest("filter for item not provided");
     }
 
-    const users = await dataAccess.retreiveFilteredUsers(req.query);
+    const { page, limit } = req.query;
+    const pageNumber = parseInt(page) || 0;
+    const limitNumber = parseInt(limit) || 100;
+    const skip = (pageNumber - 1) * limitNumber;
+
+    const users = await dataAccess.retreiveFilteredUsers(req.query, skip, limitNumber);
+    const totalCount = await users.length;
+    const totalPages = Math.ceil(totalCount / limitNumber);
+    const nextPage = pageNumber < totalPages - 1 ? pageNumber + 1 : null;
+    const prevPage = pageNumber > 0 ? pageNumber - 1 : null;
+
+    const paginationLinks = {
+      next: nextPage ? getFilteredPaginationLink(req.query, nextPage, limitNumber) : null,
+      prev: prevPage ? getFilteredPaginationLink(req.query, prevPage, limitNumber) : null,
```

Comment:
- what are we going to use this for ?

Reply:
The **paginationLinks** object in my code prepares these links(next, prev). The **getFilteredPaginationLink** function is responsible for constructing the URL for the next or previous page based on the current query parameters.

This information is useful for clients (e.g., front-end applications) that want to fetch additional pages of data without having to retrieve the entire dataset in a single request.